src/rgw/rgw_sync.cc

```diff
@@ -1322,13 +1322,14 @@ int RGWMetaSyncSingleEntryCR::operate(const DoutPrefixProvider *dpp) {
       sync_status = retcode;
 
       if (sync_status == -ENOENT) {
-        /* FIXME: do we need to remove the entry from the local zone? */
+        tn->log(10, SSTR("removing local metadata entry:" << section << ":" << key));
+        yield call(new RGWMetaRemoveEntryCR(sync_env, raw_key));
```

Comment:
what does this fix? i don't see any explanation in the commit message

doesn't this just remove retries from `RGWMetaRemoveEntryCR`, even though it still runs through that retry loop below?

these `Store`/`Remove` calls use `RGWObjVersionTracker` to detect races, so i think it's important to retry on `ECANCELED`

Reply:
you are right. the correct fix is to treat `ENOENT` on `RGWMetaRemoveEntryCR` as success